modules/dmrpp_module/build_dmrpp_h4/build_dmrpp_util_h4.cc

```diff
@@ -767,14 +767,236 @@ cout <<"length is: "<<length <<endl;
     return true;
 
 }
+
+#if 0
+bool obtain_compress_encode_data(string &encoded_str, const Bytef*source_data,size_t source_data_size, string &err_msg) {
+
+    uLong ssize = (uLong)source_data_size;
+    uLongf csize = (uLongf)ssize*2;
+    vector<Bytef> compressed_src;
+    compressed_src.resize(source_data_size*2);
+
+    int retval = compress(compressed_src.data(), &csize, source_data, ssize);
+    if (retval != 0) {
+        err_msg = "Fail to compress the data";
+        return false;
+    }
+
+    encoded_str = base64::Base64::encode(compressed_src.data(),(int)csize);
+
+    return true;
+
+}
+#endif
+
+
+bool add_missing_eos_latlon(const string &filename,BaseType *btp, const D4Attribute *eos_ll_attr, string &err_msg) {
+
+    VERBOSE(cerr<<"Coming to add_missing_eos_latlon"<<endl);
+    
+    string eos_ll_attr_value = eos_ll_attr->value(0);
+
+    VERBOSE(cerr<<"eos_ll_attr_value: "<<eos_ll_attr_value <<endl);
+
+    size_t space_pos = eos_ll_attr_value.find(' ');
+    if (space_pos ==string::npos) 
+        throw BESInternalError("Attribute eos_latlon must have space inside",__FILE__,__LINE__);
+    string grid_name = eos_ll_attr_value.substr(0,space_pos);
+    string ll_name = eos_ll_attr_value.substr(space_pos+1);
+
+    VERBOSE(cerr<<"grid_name: "<<grid_name <<endl);
+    VERBOSE(cerr<<"ll_name: "<<ll_name<<endl);
+    bool is_lat = true;
+    if (ll_name =="lon")
+        is_lat = false;
+
+    int32 gridfd = GDopen(const_cast < char *>(filename.c_str()), DFACC_READ);
```

Comment:
These const_cast<>() uses could be a problem. Are you sure the GDopen() and attach() calls don't try to write to the strings? If they do, we could change this to:

std::vector<char> name(str.begin(), str.end());
data.push_back('\0');

then 

GDopen((char *)name.data(), DFACC_READ);

With the caveat that I've not tested this, so it might contain mistakes. 

This is not needed, however, if we know the strings won't be modified.

Reply:
We have to use const_cast. I haven't tried  the way you listed. Will do that later. The file name won't be modified.